Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp

```diff
@@ -166,10 +173,36 @@ bool FontCustomPlatformData::supportsFormat(const String& format)
         || equalLettersIgnoringASCIICase(format, "svg"_s);
 }
 
-bool FontCustomPlatformData::supportsTechnology(const FontTechnology&)
+bool FontCustomPlatformData::supportsTechnology(const FontTechnology& technology)
 {
-    // FIXME: define supported technologies for this platform (webkit.org/b/256310).
-    return true;
+#if USE(HARFBUZZ)
+    // https://harfbuzz.github.io/what-does-harfbuzz-do.html
+    // Many of these features *could* be disabled but hb doesn't easily expose
+    // this and it is unlikely.
```

Comment:
I don't know where this is used, but that harfbuzz supports a feature doesn't mean WebKit uses it. Are you sure we support all the features listed below? Or are all of those used by harfbuzz internally? It's not the case of variations, for example (even though we support them, but it's just an example of a feature that requires implementation in WebKit)

Reply:
This is what chromium does FWIW:

```
bool IsSupportedKeywordTech(CSSValueID keyword) {
  switch (keyword) {
    case CSSValueID::kFeaturesOpentype:
    case CSSValueID::kFeaturesAat:
    case CSSValueID::kColorCOLRv0:
    case CSSValueID::kColorCOLRv1:
    case CSSValueID::kColorSbix:
    case CSSValueID::kColorCBDT:
    case CSSValueID::kVariations:
    case CSSValueID::kPalettes:
      return true;
    case CSSValueID::kFeaturesGraphite:
    case CSSValueID::kColorSVG:
    case CSSValueID::kIncremental:
      return false;
    default:
      return false;
  }
  NOTREACHED();
  return false;
}
```

Reply:
I tested some of them using https://opentype.dev/ and random other font sites I found.

I believe the formats are transparent (CBDT, COLR, Aat, SBix, Opentype) and I verified CBDT, COLR, and ColorSVG (of course OpenType works in general).

I tested Variations.

I didn't test Graphite, and I wasn't sure how to test Incremental/Pallets but Harfbuzz does support them.

Reply:
If we want to be conservative and move Graphite/Incremental to not supported that's fine. But ColorSVG does work.

Reply:
I just tried a font that was labeled as using Graphite and it works fine too.

Reply:
https://github.com/WebKit/WebKit/pull/23416